### PR TITLE
Followup layer metadata search widget

### PR DESCRIPTION
**Pull request by elpaso**
- Fix a bug with PG SQL update
- Handle raster icons
- Fix tab order in the widget
- Proxy model and GUI filter by geometry type

Ok, the last qualifies as small new feature sneaking in... 

immagine

Funded by: ARPA Piemonte

### CODE REVIEW

File: src/gui/qgslayermetadataresultsproxymodel.h [hunk 1]
```diff
@@ -63,6 +73,7 @@ class GUI_EXPORT QgsLayerMetadataResultsProxyModel : public QSortFilterProxyMode
 
     QgsRectangle mFilterExtent;
     QString mFilterString;
+    QString mFilterGeometryTypeName;
```
Comment (nyalldawson) [inline src/gui/qgslayermetadataresultsproxymodel.h:76]: I think using a string here is messy -- it'd be better to use the geometry type enum (with a seperate bool for rasters, or use a map layer type filter for those instead)

File: src/gui/qgslayermetadatasearchwidget.cpp [hunk 2]
```diff
@@ -48,6 +48,19 @@ QgsLayerMetadataSearchWidget::QgsLayerMetadataSearchWidget( QWidget *parent, Qt:
   mExtentFilterComboBox->addItem( QStringLiteral( "Map Canvas Extent" ) );
   mExtentFilterComboBox->addItem( QStringLiteral( "Current Project Extent" ) );
   mExtentFilterComboBox->setCurrentIndex( 0 );
+  mExtentFilterComboBox->setSizeAdjustPolicy( QComboBox::SizeAdjustPolicy::AdjustToContents );
+  mExtentFilterComboBox->adjustSize();
+
+  mGeometryTypeComboBox->addItem( QString( ), QVariant() );
```
Comment (nyalldawson) [inline src/gui/qgslayermetadatasearchwidget.cpp:54]: Can you add icons to this combo too? It makes a big usability difference

File: src/gui/qgslayermetadataresultsproxymodel.h (hunk 1 above)
Comment (elpaso) [reply src/gui/qgslayermetadataresultsproxymodel.h:76]: done!

File: src/gui/qgslayermetadatasearchwidget.cpp (hunk 2 above)
Comment (elpaso) [reply src/gui/qgslayermetadatasearchwidget.cpp:54]: done!

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**